packages/testing-lib/src/waitForTransition/waitForTransition.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { createRef } from 'react';
+import { Transition } from 'react-transition-group';
+import { render, waitFor } from '@testing-library/react';
+
+import { css } from '@leafygreen-ui/emotion';
+
+import { waitForTransition } from '.';
+
+const ref = createRef<HTMLElement>();
+
+const callbacks = {
+  onEnter: jest.fn(),
+  onEntered: jest.fn(),
+  onEntering: jest.fn(),
+  onExit: jest.fn(),
+  onExited: jest.fn(),
+  onExiting: jest.fn(),
+};
+
+const renderTransitionElement = (in0: boolean) => {
+  const result = render(
+    <Transition
+      nodeRef={ref}
+      in={in0}
+      timeout={0}
+      onEntered={callbacks.onEntered}
+      onExited={callbacks.onExited}
+    >
+      {state => (
+        <div
+          data-testid="test-div"
+          className={css`
+            opacity: ${state === 'entered' ? 1 : 0};
+          `}
+        >
+          {state}
+        </div>
+      )}
+    </Transition>,
+  );
+
+  const rerender = (in1: boolean) =>
+    result.rerender(
+      <Transition
+        nodeRef={ref}
+        in={in1}
+        timeout={0}
+        onEntered={callbacks.onEntered}
+        onExited={callbacks.onExited}
+      >
+        {state => (
+          <div
+            data-testid="test-div"
+            className={css`
+              opacity: ${state === 'entered' ? 1 : 0};
+            `}
+          >
+            {state}
+          </div>
+        )}
+      </Transition>,
+    );
+
+  return {
+    ...result,
+    rerender,
+  };
+};
+
+describe('packages/testing-lib/waitForTransition', () => {
+  afterEach(() => {
+    Object.values(callbacks).forEach(cb => cb.mockReset());
+  });
+
+  describe('triggers react-transition-group handlers', () => {
+    test('onEntered', async () => {
+      const renderResult = renderTransitionElement(false);
+      const testDiv = await renderResult.findByTestId('test-div');
+
+      /** ENTER */
+      renderResult.rerender(true);
```

Comment:
why do we have to await this in the test below but not here

Reply:
The `await` in the test below is essentially combining L81-88 here into one test